### PR TITLE
Mechanism to exclude files from check

### LINKED ISSUE #325: Mechanism to exclude files from check
Generated file WarningNames.kt should be excluded from code style check
Do we have an exclusion mechanism now?

### LINKED ISSUE #543: Write tests for maven plugin

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Logic in maven plugin
* Logic in gradle plugin
* Tests in gradle plugin
* Fixes for metrics_yml

This pull request closes #325 , also partially closes #543 

Added excludes in maven plugin in diktat self-check.
I've tested excludes in gradle plugin manually.

## Fixme
* [x] Update readmes
* [x] check if `package` in metrcs_maser actually helps, or will it not find `diktat-common`

### CODE REVIEW

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt [hunk 1]
```diff
@@ -45,8 +46,16 @@ open class DiktatJavaExecTaskBase @Inject constructor(
             if (diktatExtension.debug) {
                 add("--debug")
             }
-            add(diktatExtension.inputs.files.joinToString { it.path })
+            diktatExtension.inputs.files.forEach {
+                val pattern = project.trimRootDir(it.path)
```
Comment (akuleshov7) [inline diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:50]: Are you sure, that path (it) here is absolute path?

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt [hunk 2]
```diff
@@ -45,8 +46,16 @@ open class DiktatJavaExecTaskBase @Inject constructor(
...
+                add("\"$pattern\"")
+            }
+            diktatExtension.excludes?.files?.forEach {
+                val pattern = project.trimRootDir(it.path)
+                add("\"!$pattern\"")
+            }
         }
+        logger.lifecycle("Setting args to $args")
```
Comment (akuleshov7) [inline diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:58]: Actually I don’t like this logging. Say explicitly which arguments you are setting, not just “args”

Comment (petertrr) [reply diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:58]: Oh, it was supposed to be debug logging. I'll change the text and set the level to debug.

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt (hunk 1 above)
Comment (petertrr) [reply diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:50]: I believe, yes, this is from gradle filesystem API, all these methods use [project.file](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)) under the hood.

**Pull request merged**